### PR TITLE
refactor: semantic rules in language dropdown

### PR DESCRIPTION
**Pull request by ashraf-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/elements/typography.ts [hunk 1]
```diff
@@ -92,6 +92,11 @@ export const BaseElement = css<BaseElementProps>`
     text-align: ${(props) => props.align || 'start'};
     padding: ${(props) => props.padding || ''};
     color: ${({ color }) => (color ? `var(--color-${color})` : 'var(--color-black-3)')};
+
+    &[data-color='red'] {
+        color: var(--color-red);
```
Comment (ali-hosseini-fs) [inline src/components/elements/typography.ts:97]: why cant we use the color prop that used in line 94?

Comment (ashraf-deriv) [reply src/components/elements/typography.ts:97]: Because of w3c is complaining about the `color` attribute , we are avoid using that.

Comment (ali-hosseini-fs) [reply src/components/elements/typography.ts:97]: so in that case whats the point of line 94?
i mean i'm thinking that styled components doesn't add all of the attrs on the element unless we add them manually, is that right?

Comment (ashraf-deriv) [reply src/components/elements/typography.ts:97]: @ako Yes, That is right. 
In this particular PR we are reducing the number of w3c errors on `Attribute color not allowed on element [p](https://html.spec.whatwg.org/multipage/#the-p-element) at this point.`  
and line 94 still valid but with the cost of w3c errors. if we want to remove the all errors from w3c on this we have to get rid of the `color` attribute from <Header> component and use data- attributes instead.

**Pull request closed without merging**